File: Maximum product of two numbers - GFG/maximum-product-of-two-numbers.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;


// } Driver Code Ends
//User function template for C++
class Solution{
public:

	int maxProduct(int arr[], int n) {
	   int max=0;
	   int sec_max=0;
	   for(int i=0;i<n;i++)
	   {
	       if(max<arr[i])
	       {
	           sec_max=max;
	           max=arr[i];
	       }
	       else if(sec_max<arr[i] )
	       {
	           sec_max=arr[i];
	       }
	   }
	   return max*sec_max;
	}
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, i;
        cin >> n;
        int arr[n];
        for (i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        auto ans = ob.maxProduct(arr, n);
        cout << ans << "\n";
    }
    return 0;
}

// } Driver Code Ends

File: Maximum product of two numbers - GFG/maximum-product-of-two-numbers.py
#User function Template for python3
class Solution:

	def maxProduct(self,arr, n):
	    m=max(arr)
	    arr.remove(m)
	    n=max(arr)
	   # arr.sort()
        return m*n
            
		# code here


#{ 
 # Driver Code Starts
#Initial Template for Python 3




if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        n = int(input())
        arr = list(map(int, input().strip().split()))
        ob = Solution()
        ans = ob.maxProduct(arr, n)
        print(ans)
        tc -= 1

# } Driver Code Ends